packages/hurumap-core/src/ShareButton/ShareButton.js

```diff
@@ -0,0 +1,58 @@
+import { Box, SvgIcon } from "@mui/material";
+import React from "react";
+import {
+  TwitterShareButton,
+  LinkedinShareButton,
+  FacebookShareButton,
+  EmailShareButton,
+  WhatsappShareButton,
+  TelegramShareButton,
+  PinterestShareButton,
+} from "react-share";
+
+const componentMap = {
+  Facebook: FacebookShareButton,
+  Twitter: TwitterShareButton,
+  LinkedIn: LinkedinShareButton,
+  WhatsApp: WhatsappShareButton,
+  Email: EmailShareButton,
+  Telegram: TelegramShareButton,
+  Pinterest: PinterestShareButton,
+};
+
+const ShareButton = React.forwardRef(function ShareButton({
+  name,
+  url,
+  ButtonProps,
+  icon,
+  ...props
+}) {
+  const SocialButtonComponent = componentMap[name];
+
+  if (!SocialButtonComponent) {
+    return null;
+  }
```

Comment:
If I want to add a [Facebook Messenger](https://www.npmjs.com/package/react-share) share in say PesaYetu app, how would I do that using `name`/`ButtonProps`? Or is that impossible?